boards: st: fix sysbuild multi image flash

AlexFabre opened the pull request:
**Problem:**

When flashing a multi-image project with STLink through sysbuild, the flash utility is told to erase the whole flash between each single image flash.

Resulting in a partial flash where only the last image is effectively stored on flash...

**Correction:**

A `west flash` must not implicitly perform a mass erase on its own.

If a flash erase is required, the option has to be passed manually.

The problem is discussed in the following issue:
zephyrproject-rtos/zephyr#69582

*note: This is my first contribution to the project*

AlexFabre committed:
boards: st: fix sysbuild multi image flash

Problem:

When flashing a multi-image project with STLink through sysbuild,
the flash utility is told to erase the whole flash between each
single image flash.

Resulting in a partial flash where only the last image is effectively
stored on flash...

Correction:

A `west flash` must not implicitly perform a mass erase on its own.

If a flash erase is required, the option has to be passed manually.

The problem is discussed in the following issue:
zephyrproject-rtos/zephyr#69582

Due to CI tests errors, the correction is not applied on
eval board `b_u585i_iot02a`.

See following issue:
zephyrproject-rtos/zephyr#75164

Signed-off-by: Alex Fabre <alex.fabre@rtone.fr>

erwango commented:
@AlexFabre Thanks for pushing this. This will require an update on our test bench, but we'll do it.
This being said, please fix compliance script

erwango commented:
@gautierg-st Can you check the CI failure ? Looks related to ADC.

AlexFabre commented:
@erwango the problem is in fact related to the ADC. It's because the `vbat` node is enabled, but the DTS does not provide any ADC...

The problem is known and exposed here.

https://github.com/zephyrproject-rtos/zephyr/issues/72914#issuecomment-2122607099

My thought is it should be fixed in another PR

erwango commented:
@AlexFabre I know, hence my request to @gautierg-st to have a look as he's already assigned to #72914

AlexFabre commented:
@erwango I opened this PR #74934 to implement `adc1` on `nucleo_h533re`

erwango reviewed (dismissed):
To be rebased once https://github.com/zephyrproject-rtos/zephyr/pull/74934 is merged.

nashif commented:
@AlexFabre  please fix the CI failures

AlexFabre commented:
Hi @nashif 

As you've probably seen, the errors reported by the CI are not related in any way to anything I've changed.

The CI fails because many ST's boards have samples and tests that do not build.
- I've fixed one in  PR #74934
- I've opened another issue #75150 this morning to discuss how to solve another ST board example that does not build.

I don't think this PR should be blocked because of already present bugs.

erwango commented:
@AlexFabre One solution would be to remove b_u585i_iot02a from the patch set, since all issues relate to this board. In the meantime we can raise a specific GH issue for this and remove the `--erase` option when fixing the related bugs.

AlexFabre commented:
@erwango You're right, let's do this. 

I updated the commit message to link to this new issue.

Thank's
